### ISSUE TITLE
Limit number of recursions

### ISSUE THREAD

**Issue by Hnus-** [Vincit/objection.js#96]
Hello, 
I really like `^` thingy you can use in eager loading but is it possible to limit number of recursions somehow? In other words specify depth of returned tree. If its not possible how would you go about solving this? Just a point in the right direction would be enough.

**Comment by koskimas** [comment]
The relation expression language doesn't have a support for this, but this is something I'm considering adding to it. The syntax could be something like this: `movies.actors.children.^3`. Any suggestions?

For the time being you can do this with lodash and ES6 template strings:

```js
Person
  .query()
  .eager(`movies.actors${_.repeat('.children', 3)}`)
```

Not the prettiest, but at least it's a one-liner :smile:

**Comment by Hnus-** [comment]
That is exactly how I am doing it now :smiley: but as you say it doesn't feel right. The syntax you proposed would be perfect but I don't think I'll be able to implement something like this :disappointed: I am still reading through the code

**Comment by koskimas** [comment]
I'll implement this for the next alpha build.

**Comment by koskimas** [comment]
Implemented in 0.5.0-rc.1. Run npm install objection@next to install.

**Comment by Hnus-** [comment]
Awesome! Thank you for working very hard on this project I have tried every node ORM out there and yours is hands down the best.
